hazelcast/src/main/java/com/hazelcast/spi/Operation.java

```diff
@@ -345,6 +363,28 @@ short getFlags() {
         return flags;
     }
 
+
+    /**
+     * Called when an <tt>Exception</tt>/<tt>Error</tt> is thrown during operation execution.
+     * <p/>
+     * By default this method does nothing.
+     * Operation implementations can override this behaviour due to their needs.
+     * <p/>
+     * This method is called on node & thread that's executing the operation.
+     *
+     * @param e Exception/Error thrown during operation execution
+     */
+    public void onFailure(Throwable e) {
```

Comment:
Then I need to check `instanceOf` on every place `onFailure` is called. This also applies to most of the methods in `Operation` class. If we extract an interface for each method, I guess code will be chaos. I don't see any benefit here.